jerry-core/vm/vm.c

```diff
@@ -336,11 +334,9 @@ vm_run_global (const ecma_compiled_code_t *bytecode_p) /**< pointer to bytecode
   }
 #endif /* ENABLED (JERRY_MODULE_SYSTEM) */
 
-  return vm_run (bytecode_p,
-                 ecma_make_object_value (glob_obj_p),
-                 global_scope_p,
-                 NULL,
-                 0);
+  ecma_call_args_t call_args = ecma_op_function_make_args (glob_obj_p, ecma_make_object_value (glob_obj_p), NULL, 0);
```

Comment:
Having NULL as the argument list seems to be quite common, and it seems like a waste of stack. We already know there are no arguments from `argc`, would it be possible to not have to store the NULL pointer in this case?